### PR TITLE
Add raw_imu, scaled_imu, scaled_pressure, raw_gps

### PR DESCRIPTION
**Pull request by JonasVautherin**

### CODE REVIEW

File: protos/telemetry/telemetry.proto [hunk 1]
```diff
@@ -452,6 +509,29 @@ message GpsInfo {
...
+ *
+ * Warning: this is an advanced type! If you want the location of the drone, use
+ * the position instead. This message exposes the raw values of the GNSS sensor.
+ */
+ message RawGps {
+     uint64 timestamp_us = 1; // Timestamp in microseconds (UNIX Epoch time or time since system boot, to be inferred)
+     double latitude_deg = 2; // Latitude in degrees (WGS84, EGM96 ellipsoid)
+     double longitude_deg = 3; // Longitude in degrees (WGS84, EGM96 ellipsoid)
+     float absolute_altitude_m = 4; // Altitude AMSL (above mean sea level) in metres
+     uint32 eph = 5; // GPS HDOP horizontal dilution of position (unitless). If unknown, set to: UINT16_MAX
```
Comment (julianoes) [inline protos/telemetry/telemetry.proto:523]: Hang on. This should be called what it actually is. I think the comment is correct and not the variable name, hence it is `hdop` and I'm pretty sure it should be a float, and divide the number coming in via mavlink by 100, looking at https://github.com/PX4/PX4-Autopilot/blob/c2151cb4fafc8fdbbf136d6da4989185b0763217/src/modules/mavlink/mavlink_messages.cpp#L1393.

Unknown can then be `NaN`.

File: protos/telemetry/telemetry.proto [hunk 2]
```diff
@@ -452,6 +509,29 @@ message GpsInfo {
...
+     uint32 epv = 6; // GPS VDOP vertical dilution of position (unitless). If unknown, set to: UINT16_MAX
```
Comment (julianoes) [inline protos/telemetry/telemetry.proto:524]: Like above.

File: protos/telemetry/telemetry.proto [hunk 3]
```diff
@@ -452,6 +509,29 @@ message GpsInfo {
...
+     uint32 velocity_m_s = 7; // Ground velocity in metres per second
+     uint32 cog_cdeg = 8; // Course over ground (NOT heading, but direction of movement) in degrees * 100, 0.0..359.99 degrees. If unknown, set to: UINT16_MAX
+     double altitude_ellipsoid_m = 9; // Altitude in metres (above WGS84, EGM96 ellipsoid)
+     double horizontal_uncertainty_m = 10; // Position uncertainty in metres
+     double vertical_uncertainty_m = 11; // Altitude uncertainty in metres
+     double velocity_uncertainty_m_s = 12; // Velocity uncertainty in metres per second
+     double heading_uncertainty_deg = 13; // Heading uncertainty in degrees
```
Comment (julianoes) [inline protos/telemetry/telemetry.proto:531]: `float` or `double`?

File: protos/telemetry/telemetry.proto [hunk 4]
```diff
@@ -452,6 +509,29 @@ message GpsInfo {
...
+ */
+ message RawGps {
+     uint64 timestamp_us = 1; // Timestamp in microseconds (UNIX Epoch time or time since system boot, to be inferred)
+     double latitude_deg = 2; // Latitude in degrees (WGS84, EGM96 ellipsoid)
+     double longitude_deg = 3; // Longitude in degrees (WGS84, EGM96 ellipsoid)
+     float absolute_altitude_m = 4; // Altitude AMSL (above mean sea level) in metres
+     uint32 eph = 5; // GPS HDOP horizontal dilution of position (unitless). If unknown, set to: UINT16_MAX
+     uint32 epv = 6; // GPS VDOP vertical dilution of position (unitless). If unknown, set to: UINT16_MAX
+     uint32 velocity_m_s = 7; // Ground velocity in metres per second
+     uint32 cog_cdeg = 8; // Course over ground (NOT heading, but direction of movement) in degrees * 100, 0.0..359.99 degrees. If unknown, set to: UINT16_MAX
```
Comment (julianoes) [inline protos/telemetry/telemetry.proto:526]: Make it `float` and `degrees`.

File: protos/telemetry/telemetry.proto [hunk 5]
```diff
@@ -452,6 +509,29 @@ message GpsInfo {
...
+ * the position instead. This message exposes the raw values of the GNSS sensor.
+ */
+ message RawGps {
+     uint64 timestamp_us = 1; // Timestamp in microseconds (UNIX Epoch time or time since system boot, to be inferred)
+     double latitude_deg = 2; // Latitude in degrees (WGS84, EGM96 ellipsoid)
+     double longitude_deg = 3; // Longitude in degrees (WGS84, EGM96 ellipsoid)
+     float absolute_altitude_m = 4; // Altitude AMSL (above mean sea level) in metres
+     uint32 eph = 5; // GPS HDOP horizontal dilution of position (unitless). If unknown, set to: UINT16_MAX
+     uint32 epv = 6; // GPS VDOP vertical dilution of position (unitless). If unknown, set to: UINT16_MAX
+     uint32 velocity_m_s = 7; // Ground velocity in metres per second
```
Comment (julianoes) [inline protos/telemetry/telemetry.proto:525]: Make it `float`.

File: protos/telemetry/telemetry.proto [hunk 6]
```diff
@@ -452,6 +509,29 @@ message GpsInfo {
...
+     uint32 cog_cdeg = 8; // Course over ground (NOT heading, but direction of movement) in degrees * 100, 0.0..359.99 degrees. If unknown, set to: UINT16_MAX
+     double altitude_ellipsoid_m = 9; // Altitude in metres (above WGS84, EGM96 ellipsoid)
+     double horizontal_uncertainty_m = 10; // Position uncertainty in metres
+     double vertical_uncertainty_m = 11; // Altitude uncertainty in metres
+     double velocity_uncertainty_m_s = 12; // Velocity uncertainty in metres per second
+     double heading_uncertainty_deg = 13; // Heading uncertainty in degrees
+     uint32 yaw = 14; // Yaw in earth frame from north. Use 0 if this GPS does not provide yaw. Use 65535 if this GPS is configured to provide yaw and is currently unable to provide it. Use 36000 for north
```
Comment (julianoes) [inline protos/telemetry/telemetry.proto:532]: I would make it float and `yaw_deg`.

File: protos/telemetry/telemetry.proto [hunk 7]
```diff
@@ -603,6 +683,24 @@ message DistanceSensor {
...
+}
+
+// Scaled Pressure message type.
+message ScaledPressure {
+    uint64 timestamp_us = 1; // Timestamp (time since system boot)
+    float absolute_pressure_hpa = 2; // Absolute pressure in hPa
+    float differential_pressure_hpa = 3; // Differential pressure 1 in hPa
+    int32 temperature_deg = 4; // Absolute pressure temperature (in celcius)
+    int32 differential_pressure_temperature_deg = 5; // Differential pressure temperature (in celcius, 0 if not available)
+}
```
Comment (julianoes) [inline protos/telemetry/telemetry.proto:702]: I don't think this message is implemented for PX4. I would say there is no value implementing this message.

File: protos/telemetry/telemetry.proto [hunk 8]
```diff
@@ -664,6 +762,7 @@ message Imu {
     AngularVelocityFrd angular_velocity_frd = 2; // Angular velocity
     MagneticFieldFrd magnetic_field_frd = 3; // Magnetic field
     float temperature_degc = 4 [(mavsdk.options.default_value)="NaN"]; // Temperature
+    uint64 timestamp_us = 5; // Timestamp in microseconds
```
Comment (julianoes) [inline protos/telemetry/telemetry.proto:765]: :+1:

**Review by julianoes — CHANGES_REQUESTED** [review]

File: protos/telemetry/telemetry.proto (hunk 7 above)
Comment (JonasVautherin) [reply protos/telemetry/telemetry.proto:702]: @TSC21:  what do you think?

Comment (julianoes) [reply protos/telemetry/telemetry.proto:702]: @TSC21 ?

Comment (JonasVautherin) [reply protos/telemetry/telemetry.proto]: Decided to remove the raw pressure for now.

**Review by julianoes — APPROVED** [review]
Yes yes yes yes ship it.

**Pull request merged**